### PR TITLE
Disable all breakpoints bugfix

### PR DESCRIPTION
**Pull request by SougandhS**
Fix for #1583

### CODE REVIEW

**Comment by SougandhS** [comment]
Hi @jukzi 
Could you please check on this ?

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java [hunk 1]
```diff
@@ -37,7 +37,24 @@ public class DisableAllBreakpointsAction extends AbstractRemoveAllActionDelegate
...
 	@Override
 	protected boolean isEnabled() {
-		return DebugPlugin.getDefault().getBreakpointManager().getBreakpoints().length > 0;
+		boolean allDisabled = true;
+		IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+
+		for (IBreakpoint bp : breakpoints) {
+			try {
+				if (bp.isEnabled()) {
```
Comment (jukzi) [inline debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:46]: code would be smaller and faster if you eager return false if not enabled

**Review by jukzi — APPROVED** [review]
LGTM, could be improved but it is also fine as is

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java (hunk 1 above)
Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:45]: will change and test it

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java [hunk 2]
```diff
@@ -37,7 +37,22 @@ public class DisableAllBreakpointsAction extends AbstractRemoveAllActionDelegate
...
+					return true;
+				}
+			} catch (CoreException e) {
+				DebugUIPlugin.log(e);
+			}
+		}
+		if (breakpoints.length > 0 && allDisabled) {
```
Comment (jukzi) [inline debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:52]: allDisabled  is always true

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java [hunk 3]
```diff
@@ -37,7 +37,22 @@ public class DisableAllBreakpointsAction extends AbstractRemoveAllActionDelegate
...
+			return false;
+		}
+		return true;
```
Comment (jukzi) [inline debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:55]: i guess it should return false even if breakpoints.lenght==0

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java (hunk 2 above)
Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:52]: Missed that, will optimise further

File: debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java (hunk 3 above)
Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:55]: okay

Comment (jukzi) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:54]: Is it intentional to return true here - i.e. for no breakpoints?

Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:54]: Without conditions also work, directly returning false. I think thats simpler

Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java:54]: IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
		IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
		for (IBreakpoint bp : breakpoints) {
			try {
				if (bp.isEnabled()) {
					return true;
				}
			} catch (CoreException e) {
				DebugUIPlugin.log(e);
			}
		}
		return false;

Comment (SougandhS) [reply debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java]: Optimised further and tested.

**Comment by jukzi** [comment]
Thanks.

**Comment by SougandhS** [comment]
> Thanks.

Thanks for the analysis

**Pull request merged**